File: app/src/main/java/sample/mail/ru/httpsample/DownloadJSONAsync.java
package sample.mail.ru.httpsample;


import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONTokener;

public class DownloadJSONAsync extends HttpAsyncRequest{
    public static final String JSON_URL = "http://188.166.49.215/tech/imglist.json";

    public DownloadJSONAsync(MainActivity.RequestListener listener) {
        super(listener);
    }

    @Override
    protected String doInBackground(String... params) {
        if (params != null && params.length > 0) {
            HttpRequest request = new HttpRequest(params[0]);
            int status = request.makeRequest();

            if (status == HttpRequest.REQUEST_OK) {
                JSONTokener jtk = new JSONTokener(request.getContent());
                try {
                    JSONArray jsonArray = (JSONArray)jtk.nextValue();
                    StringBuilder builder = new StringBuilder();
                    for (int i = 0; i < jsonArray.length(); ++i) {
                        builder.append(jsonArray.getString(i)).append("\n");
                    }
                    return builder.toString();
                }
                catch (JSONException ex) {
                    ex.printStackTrace();
                    mErrorStringID = R.string.incorrect_json;
                }
            }
            else {
                mErrorStringID = request.getErrorStringId();
            }
        }
        else {
            mErrorStringID = R.string.too_few_params;
        }
        return null;
    }
}

File: app/src/main/java/sample/mail/ru/httpsample/DownloadJSONAsync.kt
package sample.mail.ru.httpsample


import org.json.JSONArray
import org.json.JSONException
import org.json.JSONTokener

class DownloadJSONAsync(listener: (Pair<String, Int>) -> Unit) : HttpAsyncRequest(listener) {

    override fun doInBackground(vararg params: String): Pair<String, Int> {
        if (params.isNotEmpty()) {
            val request = HttpRequest(params[0])
            val status = request.makeRequest()

            return if (status == HttpRequest.REQUEST_OK) {
                val jtk = JSONTokener(request.content)
                try {
                    val jsonArray = jtk.nextValue() as JSONArray
                    val builder = StringBuilder()
                    for (i in 0 until jsonArray.length()) {
                        builder.append(jsonArray.getString(i)).append("\n")
                    }
                    Pair(builder.toString(), 0)
                } catch (ex: JSONException) {
                    Pair("", R.string.incorrect_json)
                }
            } else {
                Pair("", request.errorStringId)
            }
        }
        return Pair("", R.string.too_few_params)
    }

    companion object {
        val JSON_URL = "http://188.166.49.215/tech/imglist.json"
    }
}
